lib/ts_utils/utils.py

```diff
@@ -203,7 +203,7 @@ def allowlists(distribution_name: str) -> list[str]:
 
 @functools.cache
 def get_gitignore_spec() -> pathspec.PathSpec:
-    with open(".gitignore", encoding="UTF-8") as f:
+    with Path(".gitignore").open(encoding="UTF-8") as f:
         return pathspec.PathSpec.from_lines("gitwildmatch", f.readlines())
```

Comment:
Unrelated to your above comment or this PR directly. But I just read pathspec's doc and found this:

> There is a specialized class, ``pathspec.GitIgnoreSpec``, which more closely
implements the behavior of **gitignore**. This uses ``GitWildMatchPattern``
pattern by default and handles some edge cases differently from the generic
``PathSpec`` class. ``GitIgnoreSpec`` can be used without specifying the pattern
factory::
> 
> 	>>> spec = pathspec.GitIgnoreSpec.from_lines(spec_text.splitlines())

```suggestion
        return pathspec.GitIgnoreSpec.from_lines(f.readlines())
```

Reply:
Could also omit the call to `.readlines()`. `GitIgnoreSpec.from_lines` takes an iterable of lines, so passing the file i/o object directly would work. Not sure if there's any reason to prefer one way or another, though

Reply:
You're completely right! I would assume `.readlines()` to just be a waste here since you have to both read the file, then iterate over the list again, instead of streaming it once.

`PathSpec.from_lines` simply does `patterns = [pattern_factory(line) for line in lines if line]`

Reply:
I opened https://github.com/python/typeshed/pull/13797